apps/io-services-cms-webapp/src/utils/subscription.ts

```diff
@@ -0,0 +1,64 @@
+import { ApiManagementClient } from "@azure/arm-apimanagement";
+import {
+  IResponseErrorForbiddenNotAuthorized,
+  IResponseErrorInternal,
+  IResponseErrorNotFound,
+  IResponseErrorTooManyRequests,
+  ResponseErrorForbiddenNotAuthorized,
+  ResponseErrorInternal,
+} from "@pagopa/ts-commons/lib/responses";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as TE from "fp-ts/lib/TaskEither";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+import { IConfig } from "../config";
+import { getSubscription } from "../lib/clients/apim-client";
+
+export type ErrorResponses =
+  | IResponseErrorNotFound
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorInternal
+  | IResponseErrorTooManyRequests;
+
+/**
+ * Using the **API Manage key** as 'Ocp-Apim-Subscription-Key', the Subscription relating to this key will have a name starting with "MANAGE-"
+ * and accordingly no longer equal to the serviceId.
+ *
+ * Therefore, since it is no longer possible to verify the equality *subscriptionId == serviceId*,
+ * it is necessary to verify that the owner of the subscription of the API Key is the same owner of the Subscription to which the ServiceId belongs
+ *
+ * @param config
+ * @param apimClient
+ * @param serviceId
+ * @param ownerSubscriptionId subscriptionId related to 'Ocp-Apim-Subscription-Key'
+ * @param userId APIM userId
+ * @returns
+ */
+export const serviceOwnerCheckManageTask = (
+  config: IConfig,
```

Comment:
General advice: in order to minimize the number of dependencies (parameters) passed into a method, consider passing it all the needed dependencies already configured (in this case, the configured `getSubscription` method), ie:

```ts
const getSubscriptionT = (apimClient: ApiManagementClient, config: IConfig) => (serviceId: NonEmptyString) => {
  getSubscription(
      apimClient,
      config.AZURE_APIM_RESOURCE_GROUP,
      config.AZURE_APIM,
      serviceId
    );

export const serviceOwnerCheckManageTask = (getSubscription: typeof ReturnValue<getSubscriptionT>, serviceId: NonEmptyString, ownerSubscriptionId: NonEmptyString,
  userId: NonEmptyString) => { ... getSubscription(serviceId) ... }
```

the advantages of this approach are more evident when you combine ReaderTaskEither(s), in order to pass _all_ the dependencies only once when calling the outer `ReaderTaskEither`. In order to do this you have to invert parameters order:

```ts
interface GetSubscriptionEnv {
  apimClient: ApiManagementClient;
  config: IConfig;
}

const getSubscriptionRTE =  (serviceId: NonEmptyString): 
   RTE.ReaderTaskEither<GetSubscriptionEnv, Error, Subscription> => 
({apimClient, config}: GetSubscriptionEnv) => {
  getSubscription(
      apimClient,
      config.AZURE_APIM_RESOURCE_GROUP,
      config.AZURE_APIM,
      serviceId
    );
type GetSubscriptionRTE = typeof getSubscriptionRTE;

// the following method returns a ReaderTaskEither
export const serviceOwnerCheckManage  = (serviceId: NonEmptyString, ownerSubscriptionId: NonEmptyString,
       userId: NonEmptyString) => pipe(RTE.Do, ... getSubscriptionRTE(serviceId) ... )

// ... later, on caller ...

const doSomething = (serviceId: NonEmptyString, ownerSubscriptionId: NonEmptyString,
       userId: NonEmptyString) =>

pipe(
  RTE.Do,
....
  RTE.bind("something", serviceOwnerCheckManage(serviceId, ownerSubscriptionId, 
      userId)), ...

// here you pass all the needed dependencies once:
)( {  apimClient, config  } )


```

In this way you can delete all the `makeSomethingHandler` methods and use them directly as RTE.

(not needed for this PR anyway)

Reply:
Thanks fro the tip, a ticket to make this improvement has been created